### PR TITLE
Simplified the selection of the Magisk release

### PR DESCRIPTION
**Pull request by alex-sandro92**
In the Magisk field of the workflow window, you can type "stable" (default) or "canary" to download the desired release, instead of the direct download link.  
You can alternatively put a direct download link instead of "stable" or "canary".

### CODE REVIEW

**Comment by yujincheng08** [comment]
I prefer reading from https://github.com/topjohnwu/magisk-files/blob/master/stable.json and https://github.com/topjohnwu/magisk-files/blob/master/canary.json

File: .github/workflows/magisk.yml [hunk 1]
```diff
@@ -5,9 +5,9 @@ on:
   workflow_dispatch:
     inputs:
       magisk_apk:
-        description: "Download link to magisk apk."
+        description: "Download magisk apk. Should be: [stable, canary]"
```
Comment (yujincheng08) [inline .github/workflows/magisk.yml:8]:
```suggestion
        description: "Magisk version. Should be stable, canary or a download link to Magisk apk."
```

**Review by yujincheng08 — CHANGES_REQUESTED** [review]

**Comment by yujincheng08** [comment]
How about beta version

**Comment by alex-sandro92** [comment]
> I prefer reading from https://github.com/topjohnwu/magisk-files/blob/master/stable.json and https://github.com/topjohnwu/magisk-files/blob/master/canary.json

Yeah, that was what I was looking for, much better.
I didn't consider beta version because at the moment there are only stable and canary, but it can be included anyway

**Comment by alex-sandro92** [comment]
@yujincheng08 I opened #221 by mistake, sorry. 
Anyway made the changes you suggested me there, thank you! Now it seems to be fine

File: .github/workflows/magisk.yml [hunk 2]
```diff
@@ -109,11 +109,18 @@ jobs:
...
+          from urllib.request import urlopen
 
           magisk_apk = """${{ github.event.inputs.magisk_apk }}"""
-
+                    
           if not magisk_apk:
-              magisk_apk = """https://cdn.jsdelivr.net/gh/topjohnwu/magisk-files@24.1/app-release.apk"""
+              magisk_apk = "stable"
+          if magisk_apk == "stable" or magisk_apk == "beta" or magisk_apk == "canary":
+              magisk_file = urlopen(f"https://raw.githubusercontent.com/topjohnwu/magisk-files/master/{magisk_apk}.json")
```
Comment (yujincheng08) [inline .github/workflows/magisk.yml:120]: use `requests.get`

**Review by yujincheng08 — CHANGES_REQUESTED** [review]

File: .github/workflows/magisk.yml (hunk 2 above)
Comment (alex-sandro92) [reply .github/workflows/magisk.yml]: I tried but I get <response [404]> with `requests.get`

File: .github/workflows/magisk.yml [hunk 3]
```diff
@@ -109,11 +109,18 @@ jobs:
           import urllib.request
           import zipfile
           import os
+          import json
+          from urllib.request import urlopen
```
[the remark this answers is not in the document]
Comment (alex-sandro92) [reply .github/workflows/magisk.yml]: I removed line 113 and used `urllib.request.urlopen` instead. 
Is it fine this way or do you say `requests.get` is better anyway?

[the remark this answers is not in the document]
Comment (yujincheng08) [reply .github/workflows/magisk.yml]: To unify the code. The whole code uses requests.get to obtain json.

**Pull request merged**